### ISSUE TITLE
Can't copy & paste supa_audit--0.3.1.sql to create the extension anymore

### ISSUE THREAD

**Issue by tristdrum** [supabase/supa_audit#61]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When copying and pasting the `supa_audit--0.3.1.sql` file into a SQL query on my Supabase project, I get the following error:

`ERROR:  0A000: pg_extension_config_dump() can only be called from an SQL script executed by CREATE EXTENSION`

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to [the file](https://github.com/supabase/supa_audit/blob/v0.3.1/supa_audit--0.3.1.sql)
2. Click on 'copy raw file'
3. Go to [database.new](database.new) and create your project
4. Click on SQL Editor on the sidebar (URL would be https://supabase.com/dashboard/project/project-ref-here/sql/new)
5. Paste in the code copied on step 2
6. Click run
7. See the error (`ERROR:  0A000: pg_extension_config_dump() can only be called from an SQL script executed by CREATE EXTENSION`)

## Expected behavior

I expect the extension to be created. (i.e. as if you copied and pasted the [equivalent file from v0.3.0](https://github.com/supabase/supa_audit/blob/v0.3.0/supa_audit--0.3.0.sql))

## System information

- OS: macOS
- Browser: chrome

## Additional context

Even after I [run the v0.3.0 file](https://github.com/supabase/supa_audit/blob/v0.3.0/supa_audit--0.3.0.sql), and then try to run [the patch](https://github.com/supabase/supa_audit/blob/v0.3.1/supa_audit--0.3.0--0.3.1.sql), I see get the same error.

I believe this comes from PR #54

cc: @minus7 & @olirice

**Comment by olirice** [comment]
Being able to copy/paste as an installation method was always something that incidentally worked vs officially supported.

The reason that no longer works is due to these lines

https://github.com/supabase/supa_audit/blob/4f527f062e85ec0549c45009c3860d50ddf7f282/supa_audit--0.3.1.sql#L59-L61

which is extension specific configuration to control if tables controlled by an extension should be included in backups/dumps.

If you want to copy/paste to install the extension you'll have to remove those 2 lines and everything else should just work

I'm going to close this for now but please let me know if you have any trouble getting that to work

**Comment by tristdrum** [comment]
Thanks for the quick feedback @olirice, how does one get this to work on Supabase with those lines working? (i.e. so that the data is included in backups?)

**Comment by olirice** [comment]
> how does one get this to work on Supabase with those lines working?

The standard way to install extension is using
```
make install
```
but that requires having server access.

if you install it using that mechanism, postgres recognizes it as a extension that can be created with `create extension supa_audit` and permits those 2 lines to execute sucessfully.

this extension is not supported on the Supabase platform. It is the companion code for a [blog post](https://supabase.com/blog/postgres-audit) we wrote a few years ago.

we recommend pgaudit on the supabase platform as it is more configurable and scalable

**Comment by mrjogo** [comment]
> how does one get this to work on Supabase with those lines working? (i.e. so that the data is included in backups?)

Is it correct that if `supa_audit` is installed via copy / paste, it's not technically an "extension", so the tables it creates will be included in backups even without the `pg_catalog.pg_extension_config_dump` lines?

**Comment by minus7** [comment]
Yes, if you drop the `pg_extension_config_dump` lines and copy/paste the rest it should be treated as a normal schema/table. Do note that the choice of user you use to execute the DDL statements has impact on who could manually insert/delete into the `record_version` table though.

**Comment by mrjogo** [comment]
> Do note that the choice of user you use to execute the DDL statements has impact on who could manually insert/delete into the `record_version` table though.

Do any additional changes need to be made to properly protect the `record_version` table if its executed by the default `postgres` user?

**Comment by minus7** [comment]
> Do any additional changes need to be made to properly protect the `record_version` table if its executed by the default `postgres` user?

Postgres' default default privileges grant every user (`PUBLIC`) readonly access, but best check for yourself (e.g. `\dp audit.record_version` in `psql`).

The audit function runs with `SECURITY DEFINER`, i.e. will run with the privileges of the user that created it, i.e. `postgres`, though that's the same when running `CREATE EXTENSION` as `postgres`.

tl;dr should be fine, might want to revoke read access
